x/pairing/keeper/filters/selected_providers_filter.go

```diff
@@ -0,0 +1,38 @@
+package filters
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	projectstypes "github.com/lavanet/lava/x/projects/types"
+)
+
+type SelectedProvidersFilter struct {
+	selectedProviders []string
+}
+
+func (f *SelectedProvidersFilter) InitFilter(strictestPolicy projectstypes.Policy) bool {
+	switch strictestPolicy.SelectedProvidersMode {
+	case projectstypes.Policy_EXCLUSIVE, projectstypes.Policy_MIXED:
+		f.selectedProviders = strictestPolicy.SelectedProviders
+		return true
+	}
+	return false
+}
+
+func (f *SelectedProvidersFilter) Filter(ctx sdk.Context, providers []epochstoragetypes.StakeEntry) []bool {
+	filterResult := make([]bool, len(providers))
+	if len(f.selectedProviders) == 0 {
+		return filterResult
+	}
+
+	selectedProvidersMap := map[string]string{}
```

Comment:
I'm using the map to check if the provider address exist in the providers array. I don't need to copy again the entire stake entry. It'll be wasteful since I don't need it anyway

Reply:
you misunderstood, the use of a map[string]string to check existence is wasteful a map[string]struct{}{} is more efficient and is the standard, such as this random stackoverflow: https://stackoverflow.com/questions/30213739/detecting-if-struct-exists-for-key-in-map, (you don't need to typedef)

Reply:
done. see https://github.com/lavanet/lava/pull/531/commits/67e241c199377ef08bb5058b3b2d75aaa99ce9c3